feature/amazon/src/test/java/com/revenuecat/purchases/amazon/AmazonCacheTest.kt

```diff
@@ -0,0 +1,157 @@
+        val expected = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.weekly",
+            "4321abcdreceiptid" to "com.revenuecat.subscription.monthly"
+        )
+
+        underTest.setReceiptSkus(expected)
+
+        val actualStoredJSON = JSONObject(cache.stubCache[underTest.amazonPostedTokensKey])
+        val actualStoredMapAsJSON = actualStoredJSON["receiptsToSkus"] as JSONObject
+
+        assertThat(actualStoredMapAsJSON).isNotNull
+        assertThat(actualStoredMapAsJSON.keys().asSequence().count()).isEqualTo(expected.size)
+
+        expected.forEach { (key, value) ->
+            assertThat(actualStoredMapAsJSON[key]).isEqualTo(value)
+        }
+    }
+
+    @Test
+    fun `set receipt term skus on a non empty cache`() {
+        val alreadyCached = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.weekly"
+        )
+
+        val cachedReceiptsToTermSkusJSON = getStoredJSONFromMap(alreadyCached)
+        cache.stubCache[underTest.amazonPostedTokensKey] = cachedReceiptsToTermSkusJSON
+
+        val newToCache = mapOf(
+            "4321abcdreceiptid" to "com.revenuecat.subscription.monthly"
+        )
+
+        underTest.setReceiptSkus(newToCache)
+
+        val actualStoredJSON = JSONObject(cache.stubCache[underTest.amazonPostedTokensKey])
+        val actualStoredMapAsJSON = actualStoredJSON["receiptsToSkus"] as JSONObject
+
+        assertThat(actualStoredMapAsJSON).isNotNull
+
+        val expected = alreadyCached + newToCache
+
+        assertThat(actualStoredMapAsJSON.keys().asSequence().count()).isEqualTo(expected.size)
+
+        expected.forEach { (key, value) ->
+            assertThat(actualStoredMapAsJSON[key]).isEqualTo(value)
+        }
+    }
+
+    @Test
+    fun `overriding a receipt term sku`() {
+        val alreadyCached = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.weekly"
+        )
+
+        val cachedReceiptsToTermSkusJSON = getStoredJSONFromMap(alreadyCached)
+        cache.stubCache[underTest.amazonPostedTokensKey] = cachedReceiptsToTermSkusJSON
+
+        val expected = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.monthly"
+        )
+
+        underTest.setReceiptSkus(expected)
+
+        val actualStoredJSON = JSONObject(cache.stubCache[underTest.amazonPostedTokensKey])
+        val actualStoredMapAsJSON = actualStoredJSON["receiptsToSkus"] as JSONObject
+
+        assertThat(actualStoredMapAsJSON).isNotNull
+        assertThat(actualStoredMapAsJSON.keys().asSequence().count()).isEqualTo(expected.size)
+
+        expected.forEach { (key, value) ->
+            assertThat(actualStoredMapAsJSON[key]).isEqualTo(value)
+        }
+    }
+
+    private fun getStoredJSONFromMap(expected: Map<String, String>) = """
+                { "receiptsToSkus": 
+                    {
+                        ${expected.map { "\"${it.key}\": \"${it.value}\"" }.joinToString()}
```

Comment:
kotlin is awesome.

Reply:
also, doesn't this need to be comma-separated?

Reply:
", "  is the default separator of `joinToString`

Reply:
freaking kotlin gets better the more I see of it.